src/Microsoft.DotNet.Build.Tasks/Targets/packages.targets

```diff
@@ -61,7 +61,7 @@
 
     <GetPackageVersion
       Condition="Exists('$(ToolsDir)Microsoft.DotNet.Build.Tasks.dll')"
-      DailyBuildNumber="$(DailyBuildNumber)"
+      BuildNumber="$(BuildNumber)"
```

Comment:
Thinking about this a bit more, we can condition the GetNuGetPackageVersions target to run only if BuildNumber is defined.  I think this is a bit more clean than picking some arbitrary default value.